lib/jormungandr/src/Cardano/Pool/Jormungandr/Metrics.hs

```diff
@@ -139,6 +145,48 @@ data Block = Block
     -- ^ Any stake pools that were registered in this block.
     } deriving (Eq, Show, Generic)
 
+
+-- | @StakePoolLayer@ is a thin layer ontop of the DB. It is /one/ value that
+-- can easily be passed to the API-server, where it can be used in a simple way.
+data StakePoolLayer e m = StakePoolLayer
+    { listStakePools
+        :: ExceptT e m [(StakePool, Maybe StakePoolMetadata)]
+
+    , knownStakePools
+        :: m [PoolId]
+        -- ^ Get a list of known pools that doesn't require fetching things from
+        -- any registry. This list comes from the registration certificates
+        -- that have been seen on chain.
+    }
+
+--------------------------------------------------------------------------------
+-- Api Handler
+--------------------------------------------------------------------------------
+
+listPools
+    :: LiftHandler e
+    => StakePoolLayer e IO
+    -> Handler [ApiJormungandrStakePool]
+listPools spl =
+    liftHandler $ map (uncurry mkApiJormungandrStakePool) <$> listStakePools spl
```

Comment:
Oh, yeah, I put the API handler here.

For the shelley version I was thinking that a `StakePoolLayer` _might_ be unnecessary, and that we should implement the handler directly.

Reply:
> Any strong objections or suggestions for a better place?

Yes --> 

```hs
module Cardano.Wallet.Jormungandr.Api.Server
```